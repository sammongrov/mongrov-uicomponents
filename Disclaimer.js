import React, { Component } from 'react';
import { ScrollView, View } from 'react-native';
import PropTypes from 'prop-types';
import theme, { styles } from 'react-native-theme';
import Button from './Button';
import Text from './Text';
import { translate } from '../i18n';

export default class Disclaimer extends Component {
  constructor(props) {
    super(props);
    const { locale } = this.props;
    this.locale = locale;
  }

  componentWillMount() {
    theme.setRoot(this);
  }

  render() {
    const { content, accept, onAccept } = this.props;
    return (
      <View style={[styles.fl_1, styles.p_20]}>
        <ScrollView style={[styles.primaryContent]} showsVerticalScrollIndicator={false}>
          <Text style={styles.paraText}>{content}</Text>
        </ScrollView>
        <View style={styles.paddingHorizontal80}>
          <Button
            testID="accept-button"
            onPress={onAccept}
            title={accept}
            buttonStyle={[styles.disclaimerButtonStyle]}
            containerStyle={[styles.marginTop8, styles.paddingHorizontal80]}
          />
        </View>
      </View>
    );
  }
}

Disclaimer.propTypes = {
  onAccept: PropTypes.func.isRequired,
  content: PropTypes.string.isRequired,
  accept: PropTypes.string,
  locale: PropTypes.string,
};

Disclaimer.defaultProps = {
  accept: translate('IAccept', this.locale),
  locale: null,
};
