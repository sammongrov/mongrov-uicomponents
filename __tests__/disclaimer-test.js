import React from 'react';
import { Alert } from 'react-native';

import renderer from 'react-test-renderer';
import Disclaimer from '../Disclaimer';

const onAccept = () => Alert.alert('Accepted');

it('Disclaimer renders correctly', () => {
  const tree = renderer
    .create(<Disclaimer onAccept={onAccept} content="Disclaimer is agreed" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
