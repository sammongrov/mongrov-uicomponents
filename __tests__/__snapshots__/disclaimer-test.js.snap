// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      Array [
        undefined,
      ]
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "OpenSans-Regular",
            },
            null,
          ]
        }
      >
        Disclaimer is agreed
      </Text>
    </View>
  </RCTScrollView>
  <View>
    <View
      style={
        Array [
          Array [
            undefined,
            undefined,
          ],
          false,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "borderRadius": 3,
                "flexDirection": "row",
                "justifyContent": "center",
              },
              Array [
                undefined,
              ],
              false,
              false,
              false,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                undefined,
                false,
                false,
              ]
            }
          >
            I Accept
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
